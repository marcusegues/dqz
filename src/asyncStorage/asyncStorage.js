// @flow

// $FlowFixMe
import { AsyncStorage } from 'react-native';
import type { CurrencyObject } from '../model/currencies';
import type { StoreType } from './storeTypes';

export const storeItemAsyncStorage = async (
  key: StoreType,
  item: any
): Promise<boolean> => {
  try {
    await AsyncStorage.setItem(`@Dazit:${key}`, JSON.stringify(item));
    return true;
  } catch (error) {
    // Error saving data
  }
  return false;
};

export const fetchGenericDataAsyncStorage = async (
  key: StoreType
): Promise<string> => {
  try {
    const value = await AsyncStorage.getItem(`@Dazit:${key}`);
    if (value !== null) {
      return value;
    }
  } catch (error) {
    // Error retrieving data
  }
  return '';
};

export const fetchCurrencyObjectsAsyncStorage = async (
  key: StoreType
): Promise<CurrencyObject> => {
  try {
    const value = await AsyncStorage.getItem(`@Dazit:${key}`);
    if (value !== null) {
      return JSON.parse(value);
    }
  } catch (error) {
    // Error retrieving data
  }
  return {};
};
