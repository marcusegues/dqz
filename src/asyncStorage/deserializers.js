// @flow
import Immutable from 'immutable';
import {
  makeAmountsOfCurrencyRecord,
  makeCategoryBasketItemRecord,
  makeCategoryVolumeRecord,
  makePeopleRecord,
} from '../model/types/basketPeopleAmountsTypes';
import type {
  Amounts,
  Basket,
  People,
} from '../model/types/basketPeopleAmountsTypes';

export const deserializePeople = (serialized: any): People =>
  makePeopleRecord(serialized);

export const deserializeBasket = (serialized: any): Basket => {
  const basketMap = Immutable.Map(serialized);
  return basketMap.map(d =>
    makeCategoryBasketItemRecord({
      volume: makeCategoryVolumeRecord({
        quantities: Immutable.List(d.volume.quantities),
      }),
    })
  );
};

export const deserializeAmounts = (serialized: any): Amounts => {
  const amountsMap = Immutable.Map(serialized);
  return amountsMap.map(d =>
    makeAmountsOfCurrencyRecord({
      amounts: Immutable.List(d.amounts),
      largeAmounts: Immutable.List(d.largeAmounts),
    })
  );
};

export const deserializeMainCategories = (serialized: string): any =>
  Immutable.Set(serialized);
