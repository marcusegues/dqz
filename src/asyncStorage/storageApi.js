// @flow

import {
  fetchAmountsAsyncStorage,
  fetchBasketAsyncStorage,
  fetchCurrencyObjectsAsyncStorage,
  fetchPeopleAsyncStorage,
  fetchSettingsAcceptRateAsyncStorage,
  fetchSettingsHasLanguageAsyncStorage,
  storeItemAsyncStorage,
} from './asyncStorage';
import type { CurrencyObject } from '../model/currencies';
import type { StoreType } from './storeTypes';
import type { Language } from '../i18n/types/locale';
import type {
  Amounts,
  Basket,
  People,
} from '../model/types/basketPeopleAmountsTypes';

/**
 * Stores item (stringified) under key - do NOT use directly!
 * @param key
 * @param item
 * @returns {Promise<boolean>}
 */
export const storeItem = async (key: StoreType, item: any): Promise<boolean> =>
  storeItemAsyncStorage(key, item);

/**
 * Store a currency object
 * @param currencyObject
 * @returns {Promise<boolean>}
 */
export const storeCurrencyObject = (currencyObject: CurrencyObject) =>
  storeItem('CurrencyObject', currencyObject);

/**
 * Sets a flag if the user has a language set
 * @param flag
 * @returns {Promise<boolean>}
 */
export const storeSettingsAcceptRate = (flag: boolean) =>
  storeItem('SettingsAcceptRate', flag);

/**
 * Sets a flag if the user has a language pre-chosen
 * @param flag
 * @returns {Promise<boolean>}
 */
export const storeSettingsHasLanguage = (flag: Language) =>
  storeItem('SettingsHasLanguage', flag);

/**
 * Stores a basket
 * @param basket
 * @returns {Promise<boolean>}
 */
export const storeBasket = (basket: Basket) =>
  storeItem('Basket', basket.toJS());

/**
 * Stores amounts
 * @param amounts
 * @returns {Promise<boolean>}
 */
export const storeAmounts = (amounts: Amounts) =>
  storeItem('Amounts', amounts.toJS());

/**
 * Stores people
 * @param people
 * @returns {Promise<boolean>}
 */
export const storePeople = (people: People) =>
  storeItem('People', people.toJS());

/**
 * Fetch a currency object
 * @returns {Promise<CurrencyObject>}
 */
export const fetchCurrencyObject = async (): Promise<CurrencyObject> =>
  fetchCurrencyObjectsAsyncStorage('CurrencyObject');

/**
 * Fetches the accept-rate flag
 * @returns {Promise<CurrencyObject>}
 */
export const fetchSettingsAcceptRate = async (): Promise<boolean> =>
  fetchSettingsAcceptRateAsyncStorage('SettingsAcceptRate');

/**
 * Fetches the language-set flag
 * @returns {Promise<CurrencyObject>}
 */
export const fetchSettingsHasLanguage = async (): Promise<Language> =>
  fetchSettingsHasLanguageAsyncStorage('SettingsHasLanguage');

export const fetchBasket = async (): Promise<Basket> =>
  fetchBasketAsyncStorage('Basket');

export const fetchAmounts = async (): Promise<Amounts> =>
  fetchAmountsAsyncStorage('Amounts');

export const fetchPeople = async (): Promise<People> =>
  fetchPeopleAsyncStorage('People');
