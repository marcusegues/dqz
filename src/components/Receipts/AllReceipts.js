// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import type { ComponentType } from 'react';
import type { List } from 'immutable';
import Immutable from 'immutable';
import { DateTime } from 'luxon';
import { connect } from 'react-redux';
// $FlowFixMe
import { translate } from 'react-i18next';
import { ScrollViewCard } from './subcomponents/ScrollViewCard';
import type { Navigation, TFunction } from '../../types/generalTypes';
import { analyticsScreenMounted } from '../../analytics/analyticsApi';
import { CardHeader } from '../QuestionAnswer/cards/subcomponents/CardHeader';
import { AllReceiptsRow } from './subcomponents/AllReceiptsRow';
import { HeaderTitle } from '../Headers/subcomponents/HeaderTitle';
import { verticalScale } from '../../styles/Scaling';
import type { Receipt } from '../../types/receiptTypes';
import { fetchReceipts } from '../../asyncStorage/storageApi';
import { calculateDuty } from '../../model/dutyCalculations';
import { calculateVat } from '../../model/vatCalculations';
import { getConvertedLocalTimeToSwiss } from '../../model/utils';

type AllReceiptsState = {
  receipts: List<Receipt>,
};

type AllReceiptsProps = {
  t: TFunction,
  navigation: Navigation,
  setReceiptId: (receiptId: string) => void,
};

class AllReceiptsInner extends React.Component<
  AllReceiptsProps,
  AllReceiptsState
> {
  static navigationOptions = ({ screenProps }) => ({
    headerTitle: (
      <HeaderTitle
        text={screenProps.t('receipt:allReceiptsNavigationHeaderTitle')}
      />
    ),
  });

  constructor(props: AllReceiptsProps & { t: TFunction }) {
    super(props);
    this.state = {
      receipts: Immutable.List(),
    };
  }

  componentWillMount() {
    analyticsScreenMounted('AllReceipts');
  }

  componentDidMount() {
    fetchReceipts().then(receipts => {
      this.setState({ receipts });
    });
  }

  prepareReceiptsObject() {
    const { t, navigation, setReceiptId } = this.props;

    const sortDateTimeAsc = (a, b) =>
      DateTime.fromISO(a.receiptEntryTime) >
      DateTime.fromISO(b.receiptEntryTime)
        ? 1
        : -1;

    return this.state.receipts.sort(sortDateTimeAsc).reduce(
      (receipts, receipt) => {
        const { basket } = receipt;
        const vatReport = calculateVat(
          receipt.amounts,
          receipt.people,
          receipt.currencies
        );
        const dutyReport = calculateDuty(basket, receipt.people);
        const fullVat = vatReport.get('totalVat');
        const fullDuty = dutyReport.get('totalDuty');
        const receiptEntryTimePlus = DateTime.fromISO(
          receipt.receiptEntryTime,
          { zone: 'Europe/Zurich' }
        ).plus({ hours: 2 });
        const localSwissTime = getConvertedLocalTimeToSwiss();

        const receiptView = (
          <AllReceiptsRow
            key={`receipt-row-${receipt.receiptId}`}
            sum={t('allReceiptsSumInFranks', {
              value: (fullVat + fullDuty).toFixed(2),
            })}
            duty={t('dutyAndVat', {
              duty: fullDuty.toFixed(2),
              vat: fullVat.toFixed(2),
            })}
            date={t('allReceiptsDate', {
              value: receiptEntryTimePlus.toFormat('dd.MM.y HH:mm:ss'),
            })}
            rowOnPress={() => {
              setReceiptId(receipt.receiptId);
              navigation.navigate('ReceiptAfterPayment');
            }}
          />
        );
        if (receiptEntryTimePlus > localSwissTime) {
          // $FlowFixMe
          receipts.actualReceipts.push(receiptView);
        } else {
          // $FlowFixMe
          receipts.oldReceipts.push(receiptView);
        }
        return receipts;
      },
      { actualReceipts: [], oldReceipts: [] }
    );
  }

  render() {
    const { t } = this.props;

    // $FlowFixMe
    const sortedReceipts: {
      actualReceipts: Array<any>,
      oldReceipts: Array<any>,
    } = this.prepareReceiptsObject();

    return (
      <ScrollViewCard>
        <CardHeader text={t('allReceiptsCurrentReceipt')} />
        {sortedReceipts.actualReceipts}
        <View style={{ marginTop: verticalScale(30) }}>
          <CardHeader text={t('allReceiptsOlderReceipts')} />
        </View>
        {sortedReceipts.oldReceipts}
      </ScrollViewCard>
    );
  }
}
const mapDispatchToProps = dispatch => ({
  setReceiptId: (receiptId: string) =>
    dispatch({
      type: 'SET_RECEIPT_ID',
      receiptId,
    }),
});

export const AllReceipts = (connect(null, mapDispatchToProps)(
  translate(['receipt'])(AllReceiptsInner)
): ComponentType<AllReceiptsProps>);
