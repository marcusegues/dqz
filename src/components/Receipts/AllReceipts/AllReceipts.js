// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import type { ComponentType } from 'react';
import type { List } from 'immutable';
import Immutable from 'immutable';
import { DateTime } from 'luxon';
import { connect } from 'react-redux';
// $FlowFixMe
import { translate } from 'react-i18next';
import type { Navigation, TFunction } from '../../../types/generalTypes';
import { analyticsScreenMounted } from '../../../analytics/analyticsApi';
import { CardHeader } from '../../QuestionAnswer/Cards/subcomponents/CardHeader';
import { AllReceiptsRow } from './subcomponents/AllReceiptsRow';
import { HeaderTitle } from '../../Headers/subcomponents/HeaderTitle';
import { verticalScale } from '../../../styles/Scaling';
import type { Receipt } from '../../../types/receiptTypes';
import { fetchReceipts } from '../../../asyncStorage/storageApi';
import { calculateDuty } from '../../../model/dutyCalculations';
import { calculateVat } from '../../../model/vatCalculations';
import { getConvertedLocalTimeToSwiss } from '../../../model/utils';
import { ScrollViewCard } from '../../General Components/ScrollViewCard';
import { MainContentContainer } from '../../MainContentContainer/MainContentContainer';

type AllReceiptsState = {
  receipts: List<Receipt>,
};

type AllReceiptsProps = {
  t: TFunction,
  i18n: { language: string },
  navigation: Navigation,
  setReceiptId: (receiptId: string) => void,
};

class AllReceiptsInner extends React.Component<
  AllReceiptsProps,
  AllReceiptsState
> {
  static navigationOptions = ({ screenProps }) => ({
    headerTitle: (
      <HeaderTitle
        text={screenProps.t('receipt:allReceiptsNavigationHeaderTitle')}
      />
    ),
  });

  constructor(props: AllReceiptsProps & { t: TFunction }) {
    super(props);
    this.state = {
      receipts: Immutable.List(),
    };
  }

  componentWillMount() {
    analyticsScreenMounted('AllReceipts');
  }

  componentDidMount() {
    fetchReceipts().then(receipts => {
      this.setState({ receipts });
    });
  }

  prepareReceiptsObject() {
    const { t, i18n, navigation, setReceiptId } = this.props;

    const sortDateTimeAsc = (a, b) => {
      const aDT: DateTime = DateTime.fromISO(a.receiptEntryTime);
      const bDT: DateTime = DateTime.fromISO(b.receiptEntryTime);
      // $FlowFixMe
      return aDT > bDT ? 1 : -1;
    };

    return this.state.receipts.sort(sortDateTimeAsc).reduce(
      (receipts, receipt) => {
        const { basket } = receipt;
        const vatReport = calculateVat(
          receipt.amounts,
          receipt.people,
          receipt.currencies
        );
        const dutyReport = calculateDuty(basket, receipt.people);
        const fullVat = vatReport.get('totalVat');
        const fullDuty = dutyReport.get('totalDuty');
        const receiptEntryTimePlus = DateTime.fromISO(
          receipt.receiptEntryTime,
          { zone: 'Europe/Zurich' }
        ).plus({ hours: 2 });
        const localSwissTime = getConvertedLocalTimeToSwiss();

        const receiptView = (
          <AllReceiptsRow
            key={`receipt-row-${receipt.receiptId}`}
            sum={t('allReceiptsSumInFranks', {
              value: (fullVat + fullDuty).toFixed(2),
            })}
            duty={t('dutyAndVat', {
              duty: fullDuty.toFixed(2),
              vat: fullVat.toFixed(2),
            })}
            date={t('allReceiptsDate', {
              value:
                i18n.language === 'fr'
                  ? receiptEntryTimePlus.toFormat("dd.MM.y HH'h'mm")
                  : receiptEntryTimePlus.toFormat('dd.MM.y HH:mm'),
            })}
            rowOnPress={() => {
              setReceiptId(receipt.receiptId);
              navigation.navigate('ReceiptAfterPayment');
            }}
          />
        );
        // $FlowFixMe
        if (receiptEntryTimePlus > localSwissTime) {
          // $FlowFixMe
          receipts.actualReceipts.push(receiptView);
        } else {
          // $FlowFixMe
          receipts.oldReceipts.push(receiptView);
        }
        return receipts;
      },
      { actualReceipts: [], oldReceipts: [] }
    );
  }

  render() {
    const { t } = this.props;

    // $FlowFixMe
    const sortedReceipts: {
      actualReceipts: Array<any>,
      oldReceipts: Array<any>,
    } = this.prepareReceiptsObject();

    return (
      <MainContentContainer>
        <ScrollViewCard>
          <CardHeader text={t('allReceiptsCurrentReceipt')} />
          {sortedReceipts.actualReceipts}
          <View style={{ marginTop: verticalScale(30) }}>
            <CardHeader text={t('allReceiptsOlderReceipts')} />
          </View>
          {sortedReceipts.oldReceipts}
        </ScrollViewCard>
      </MainContentContainer>
    );
  }
}
const mapDispatchToProps = dispatch => ({
  setReceiptId: (receiptId: string) =>
    dispatch({
      type: 'SET_RECEIPT_ID',
      receiptId,
    }),
});

export const AllReceipts = (connect(null, mapDispatchToProps)(
  translate(['receipt'])(AllReceiptsInner)
): ComponentType<AllReceiptsProps>);
