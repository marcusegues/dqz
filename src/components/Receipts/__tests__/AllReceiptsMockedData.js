import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Immutable from 'immutable';
import { AllReceipts } from '../AllReceipts/AllReceipts';
import { fetchReceipts } from '../../../asyncStorage/storageApi';
import { deserializeReceipts } from '../../../asyncStorage/deserializers';

const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

const initialState = {
  receipts: Immutable.List(),
};

jest.mock('react-i18next', () => ({
  translate: () => Component => props => <Component t={() => ''} {...props} />,
}));

jest.mock('../../../asyncStorage/storageApi', () => ({
  fetchReceipts: jest.fn(),
}));
fetchReceipts.mockImplementation(
  () =>
    new Promise(resolve =>
      resolve(
        deserializeReceipts([
          {
            amounts: {
              EUR: {
                amounts: [
                  {
                    amount: 444,
                    id: '6dbd4271-bd13-46f5-916e-233dab7c6fe5',
                  },
                ],
                largeAmounts: [],
              },
            },
            basket: {
              AlcHard: {
                volume: {
                  quantities: [],
                },
              },
              AlcSoft: {
                volume: {
                  quantities: [],
                },
              },
              AnimalFeed: {
                volume: {
                  quantities: [],
                },
              },
              Books: {
                volume: {
                  quantities: [],
                },
              },
              Butter: {
                volume: {
                  quantities: [],
                },
              },
              Cigarettes: {
                volume: {
                  quantities: [],
                },
              },
              Fertilizer: {
                volume: {
                  quantities: [],
                },
              },
              Flowers: {
                volume: {
                  quantities: [],
                },
              },
              Magazines: {
                volume: {
                  quantities: [],
                },
              },
              Meat: {
                volume: {
                  quantities: [],
                },
              },
              Meds: {
                volume: {
                  quantities: [],
                },
              },
              Oils: {
                volume: {
                  quantities: [],
                },
              },
              Other: {
                volume: {
                  quantities: [],
                },
              },
              OtherFood: {
                volume: {
                  quantities: [],
                },
              },
              Tobacco: {
                volume: {
                  quantities: [],
                },
              },
            },
            currencies: {
              AED: 0.2631566,
              ALL: 0.0092876,
              ARS: 0.04788,
              AUD: 0.74573,
              AZN: 0.57872,
              BAM: 0.6090296,
              BDT: 0.011627400000000001,
              BGN: 0.6090679,
              BHD: 2.57656,
              BRL: 0.2825107,
              CAD: 0.76166,
              CHF: 1,
              CLP: 0.00160065,
              CNY: 0.15331039999999999,
              COP: 0.000348177,
              CRC: 0.00171214,
              CZK: 0.047020099999999995,
              DKK: 0.1598426,
              DOP: 0.0195642,
              EGP: 0.054740399999999995,
              ETB: 0.0354824,
              EUR: 1.19027,
              GBP: 1.36532,
              GTQ: 0.1306198,
              HKD: 0.12311899999999999,
              HNL: 0.0408636,
              HRK: 0.16054569999999999,
              HUF: 0.00382563,
              IDR: 0.000070268,
              ILS: 0.27517,
              INR: 0.0148951,
              ISK: 0.0098438,
              JPY: 0.0090377,
              KES: 0.009569,
              KHR: 0.00024156200000000002,
              KRW: 0.0009054,
              KWD: 3.23201,
              KYD: 1.17862,
              KZT: 0.00301682,
              LBP: 0.000640471,
              LKR: 0.00624092,
              LYD: 0.72667,
              MAD: 0.10567,
              MUR: 0.0298477,
              MXN: 0.05281,
              MYR: 0.2501214,
              NGN: 0.00269211,
              NOK: 0.1241705,
              NZD: 0.70722,
              OMR: 2.5156,
              PAB: 0.96647,
              PEN: 0.2987078,
              PHP: 0.0185824,
              PKR: 0.0083525,
              PLN: 0.2839049,
              QAR: 0.26547699999999996,
              RON: 0.2556661,
              RSD: 0.010066299999999999,
              RUB: 0.0159255,
              SAR: 0.2577113,
              SEK: 0.1156551,
              SGD: 0.73742,
              THB: 0.030926999999999996,
              TND: 0.401191,
              TRY: 0.2371819,
              TWD: 0.0331141,
              TZS: 0.00042745,
              UAH: 0.0376666,
              USD: 0.9664,
              UYU: 0.0344552,
              VEF: 0.000019,
              VND: 0.000042470000000000005,
              ZAR: 0.08007,
            },
            paymentData: {
              redirectUrl:
                'https://test.saferpay.com/vt2/api/PaymentPage/242565/17875435/6x0w3xkuclvukxmwr47ihqk08',
              requestId: 'ec7f12b0-3d7c-11e8-a4ed-a7739dcf82a3',
              specVersion: '1.8',
              status: 'started',
              token: '6x0w3xkuclvukxmwr47ihqk08',
              tokenExpiration: '2018-04-11T14:38:59.400+02:00',
              transaction: {
                amountValue: '4065',
                brandName: 'VISA',
                cardHolderName: 'Yamada Taro',
                cardNumber: 'xxxxxxxxxxxx0004',
                currencyCode: 'CHF',
                date: '2018-04-11T13:39:17.917+02:00',
                id: 'K5Y2OnA5W477UA7nhf9IAMz5vKIb',
                ipAddress: '81.198.116.84',
                ipLocation: 'LV',
                paymentMethod: 'VISA',
                status: 'AUTHORIZED',
              },
            },
            people: {
              adults: 1,
              minors: 0,
            },
            receiptEntryTime: '2018-04-11T14:34:59.763+03:00',
            receiptId: 'f8c48d70-3d7c-11e8-a4ed-a7739dcf82a3',
          },
          {
            amounts: {
              EUR: {
                amounts: [
                  {
                    amount: 456,
                    id: '4795ded5-3609-43e7-815c-a3cb3eac966f',
                  },
                  {
                    amount: 123,
                    id: '1bb526b5-d2a7-4309-8a92-9de9599025ca',
                  },
                ],
                largeAmounts: [],
              },
            },
            basket: {
              AlcHard: {
                volume: {
                  quantities: [],
                },
              },
              AlcSoft: {
                volume: {
                  quantities: [],
                },
              },
              AnimalFeed: {
                volume: {
                  quantities: [],
                },
              },
              Books: {
                volume: {
                  quantities: [],
                },
              },
              Butter: {
                volume: {
                  quantities: [
                    {
                      date: '2018-04-12T12:03:03.811+03:00',
                      number: 5.99,
                    },
                    {
                      date: '2018-04-12T12:03:07.647+03:00',
                      number: 0.01,
                    },
                    {
                      date: '2018-04-12T12:03:22.846+03:00',
                      number: 0.001,
                    },
                  ],
                },
              },
              Cigarettes: {
                volume: {
                  quantities: [],
                },
              },
              Fertilizer: {
                volume: {
                  quantities: [],
                },
              },
              Flowers: {
                volume: {
                  quantities: [],
                },
              },
              Magazines: {
                volume: {
                  quantities: [],
                },
              },
              Meat: {
                volume: {
                  quantities: [
                    {
                      date: '2018-04-12T12:02:50.497+03:00',
                      number: 1.7,
                    },
                    {
                      date: '2018-04-12T12:02:55.553+03:00',
                      number: 3.56,
                    },
                  ],
                },
              },
              Meds: {
                volume: {
                  quantities: [],
                },
              },
              Oils: {
                volume: {
                  quantities: [],
                },
              },
              Other: {
                volume: {
                  quantities: [],
                },
              },
              OtherFood: {
                volume: {
                  quantities: [],
                },
              },
              Tobacco: {
                volume: {
                  quantities: [],
                },
              },
            },
            currencies: {
              AED: 0.2632116,
              ALL: 0.009285999999999999,
              ARS: 0.04799,
              AUD: 0.74994,
              AZN: 0.57884,
              BAM: 0.611005,
              BDT: 0.0116889,
              BGN: 0.610619,
              BHD: 2.57724,
              BRL: 0.2836809,
              CAD: 0.76726,
              CHF: 1,
              CLP: 0.00160897,
              CNY: 0.1539015,
              COP: 0.000349016,
              CRC: 0.0017186200000000001,
              CZK: 0.047229799999999995,
              DKK: 0.1603577,
              DOP: 0.0195682,
              EGP: 0.054751799999999996,
              ETB: 0.035489799999999995,
              EUR: 1.19433,
              GBP: 1.37017,
              GTQ: 0.1305942,
              HKD: 0.12313919999999999,
              HNL: 0.040868299999999996,
              HRK: 0.16119779999999997,
              HUF: 0.0038382800000000003,
              IDR: 0.000070349,
              ILS: 0.27679,
              INR: 0.014877499999999998,
              ISK: 0.00986,
              JPY: 0.0090257,
              KES: 0.0095663,
              KHR: 0.000241612,
              KRW: 0.0009067999999999999,
              KWD: 3.23366,
              KYD: 1.17887,
              KZT: 0.0029509500000000004,
              LBP: 0.000639754,
              LKR: 0.006237390000000001,
              LYD: 0.72682,
              MAD: 0.106,
              MUR: 0.0298816,
              MXN: 0.053,
              MYR: 0.25007660000000004,
              NGN: 0.0026852,
              NOK: 0.1239974,
              NZD: 0.71171,
              OMR: 2.51613,
              PAB: 0.96667,
              PEN: 0.29844740000000003,
              PHP: 0.0186149,
              PKR: 0.0083528,
              PLN: 0.2851537,
              QAR: 0.2655325,
              RON: 0.2568406,
              RSD: 0.0101243,
              RUB: 0.0153208,
              SAR: 0.2577721,
              SEK: 0.11688330000000001,
              SGD: 0.73848,
              THB: 0.030993,
              TND: 0.40524479999999996,
              TRY: 0.2347485,
              TWD: 0.0331426,
              TZS: 0.00042753999999999996,
              UAH: 0.0376804,
              USD: 0.96677,
              UYU: 0.0343522,
              VEF: 0.000019,
              VND: 0.000042481,
              ZAR: 0.0804,
            },
            paymentData: {
              redirectUrl:
                'https://test.saferpay.com/vt2/api/PaymentPage/242565/17875435/jmfa55y3sz7jr9t7ryoixrnpk',
              requestId: '68813290-3e30-11e8-8d55-e733ef170bd1',
              specVersion: '1.8',
              status: 'started',
              token: 'jmfa55y3sz7jr9t7ryoixrnpk',
              tokenExpiration: '2018-04-12T12:03:47.368+02:00',
              transaction: {
                amountValue: '20790',
                brandName: 'VISA',
                cardHolderName: 'Yamada Taro',
                cardNumber: 'xxxxxxxxxxxx0004',
                currencyCode: 'CHF',
                date: '2018-04-12T11:03:59.723+02:00',
                id: 'Ev8OA7b8bE7xvAr90Q4Ub20vb0Ib',
                ipAddress: '81.198.116.84',
                ipLocation: 'LV',
                paymentMethod: 'VISA',
                status: 'AUTHORIZED',
              },
            },
            people: {
              adults: 1,
              minors: 0,
            },
            receiptEntryTime: '2018-04-11T13:03:00.000+03:00',
            receiptId: '70f9c6d0-3e30-11e8-8d55-e733ef170bd1',
          },
        ])
      )
    )
);

describe('AllReceipts with mocked data', () => {
  test('Renders empty AllReceipts component to snapshot', async () => {
    const tree = await renderer.create(
      <AllReceipts store={mockStore(initialState)} i18n={{ language: 'de' }} />
    );
    const json = await tree.toJSON();
    expect(json).toMatchSnapshot();
  });
});
