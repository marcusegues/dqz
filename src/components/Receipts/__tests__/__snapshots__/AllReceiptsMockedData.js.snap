// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllReceipts with mocked data Renders empty AllReceipts component to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "height": "100%",
      "marginHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#E0E0E1",
            "borderRadius": 3,
            "width": "100%",
          },
          Object {
            "elevation": 2,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 2,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "alignSelf": "flex-start",
              "marginBottom": 31.388235294117646,
              "marginHorizontal": 31.388235294117646,
              "marginTop": 31.388235294117646,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#141414",
                "fontFamily": "roboto_regular",
                "fontSize": 24,
                "lineHeight": 62.77647058823529,
              }
            }
          >
            
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderBottomWidth": 1,
                "borderColor": "#E0E0E1",
                "borderTopWidth": 0,
                "flexDirection": "column",
                "justifyContent": "space-around",
                "width": "95%",
              },
              Object {},
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingHorizontal": 12,
                  "paddingVertical": 8,
                },
              ]
            }
            testID={undefined}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                    "paddingRight": 1,
                  }
                }
              >
                <View
                  style={Object {}}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#24253D",
                        "fontFamily": "roboto_medium",
                        "fontSize": 22,
                        "paddingBottom": 3,
                      }
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#898989",
                      "fontFamily": "roboto_regular",
                      "fontSize": 18.857142857142858,
                    }
                  }
                >
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#1A1A1A",
                      "fontFamily": "roboto_regular",
                      "fontSize": 18.857142857142858,
                    }
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginTop": 58.85294117647059,
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "marginBottom": 31.388235294117646,
                "marginHorizontal": 31.388235294117646,
                "marginTop": 31.388235294117646,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#141414",
                  "fontFamily": "roboto_regular",
                  "fontSize": 24,
                  "lineHeight": 62.77647058823529,
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "borderBottomWidth": 1,
                "borderColor": "#E0E0E1",
                "borderTopWidth": 1,
                "flexDirection": "column",
                "justifyContent": "space-around",
                "width": "95%",
              },
              Object {},
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingHorizontal": 12,
                  "paddingVertical": 8,
                },
              ]
            }
            testID={undefined}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "flex-start",
                    "paddingRight": 1,
                  }
                }
              >
                <View
                  style={Object {}}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#24253D",
                        "fontFamily": "roboto_medium",
                        "fontSize": 22,
                        "paddingBottom": 3,
                      }
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#898989",
                      "fontFamily": "roboto_regular",
                      "fontSize": 18.857142857142858,
                    }
                  }
                >
                  
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#1A1A1A",
                      "fontFamily": "roboto_regular",
                      "fontSize": 18.857142857142858,
                    }
                  }
                >
                  
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
